test/controllers/matches.js

```diff
@@ -0,0 +1,172 @@
+          .get('/games/2/match')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            response.body.should.be.a('array');
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      });
+    });
+    it('should be successful getting the list with 2 matches', (done) => {
+      successfulLogin().then((res) => {
+        chai.request(server)
+          .get('/games/2/match?limit=2')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            response.body.should.be.a('array');
+            response.body.should.have.lengthOf(2);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should be successful getting the list with only 1 match because of offset', (done) => {
+      successfulLogin().then((res) => {
+        chai.request(server)
+          .get('/games/2/match?offset=1')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            response.body.should.be.a('array');
+            response.body.should.have.lengthOf(1);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should fail because of not being authenticated', (done) => {
+      chai.request(server)
+        .get('/games/2/match')
+        .send()
+        .catch((err) => {
+          err.response.should.be.json;
+          err.response.body.should.have.property('error');
+          err.response.body.error.should.equal('User is not logged in');
+          err.should.have.status(401);
+        })
+        .then((response) => {
+          done();
+        });
+    });
+  });
+  describe('/users/:userId/matches/emailSummary GET', () => {
+    it('should be successful getting the list of matches by user', (done) => {
+      successfulLogin().then((res) => {
+        chai.request(server)
+          .get('/users/1/matches/emailSummary')
+          .set(sessionManager.HEADER_NAME, res.headers.authorization)
+          .send()
+          .then((response) => {
+            response.should.have.status(200);
+            response.body.should.be.a('array');
+            response.body.should.have.lengthOf(2);
+            dictum.chai(response);
+          })
+          .then(() => {
+            done();
+          });
+      }).catch((err) => {
+        done(err);
+      });
+    });
+    it('should fail because user is not admin and is requesting for anothed user_id', (done) => {
```

Comment:
We should also check for the case when the user is not admin but askes for its own id